src/core/dataloader/utils/findByBatchIds.ts

```diff
@@ -0,0 +1,57 @@
+import { Repository } from 'typeorm';
+import { EntityNotFoundException } from '@common/exceptions';
+import { LogContext } from '@common/enums';
+import { EntityRelations } from '@src/types';
+
+const parentAlias = 'parent';
+const resultAlias = 'relation';
+
+export const findByBatchIds = async <
+  TParent extends { id: string } & { [key: string]: any }, // todo better type
+  TResult
+>(
+  repo: Repository<TParent>,
+  ids: string[],
+  relation: EntityRelations<TParent>,
```

Comment:
```suggestion
  relation: EntityRelations<TParent>[],
```

Reply:
This is addressed with a better, typed approach in the latest changes.